mass_mailing_unique/models/mail_mass_mailing_contact.py

```diff
@@ -16,9 +16,13 @@ def _check_email_list_ids(self):
             others = lists.mapped('contact_ids') - contact
 
             contact_email = contact.email.strip().lower()
-            other_emails = [e.strip().lower() for e in others.mapped('email')]
+            other_emails = {e.strip().lower() for e in others.mapped('email')}
             if contact_email in other_emails:
-                raise ValidationError(_(
-                    "Cannot have the same email (%s) more"
-                    "than once in the same list." % contact_email
-                ))
+                in_list = others.filtered(lambda o: o.email == contact_email) \
+                    .list_ids.mapped('name')
+                lists = ''.join(map(lambda l: '- {}\n'.format(l), in_list))
+                msg = _('''Email (%s) already in mailing list(s):
+                    %s
+                    Please use a different email address or remove (%s) from the \
+                    mailing list(s) above.''')
+                raise ValidationError(msg % (contact_email, lists, contact_email))
```

Comment:
Use named placeholders and a dict, to allow easier translations.

Reply:
I'm confused because you said earlier, translations must use [`%s`](https://github.com/OCA/social/pull/619/files/60b0908e2653ed66b49123dac0c174271f1917cc#r531699386)

Reply:
placeholder dicts `%(key)s` is another way for the same formatting tool. The problem is to use `.format()`.

Reply:
implemented and pushed

Reply:
Yes, there is actually another problem related, which happens when you have more than 1 placeholder in your translation. It is explained in https://github.com/OCA/pylint-odoo/issues/303#issue-758477562, in case you want more details.

Reply:
Thanks, that's just more info on why placeholder dicts should be used, right?  Nothing needs changed at this point, correct?

Reply:
Yes